libfwupdplugin/fu-backend.c

```diff
@@ -146,6 +149,38 @@ fu_backend_registered(FuBackend *self, FuDevice *device)
 		klass->registered(self, device);
 }
 
+/**
+ * fu_backend_get_device_parent:
```

Comment:
I don't feel this is named appropriately for what it's doing.  It's more like "find parent with type"

Reply:
in the USB case it's really "find the parent device" and in the UDEV case it's find the parent device with this optional udev subsystem+devtype filter. Maybe `fu_backend_get_device_parent_of_kind` perhaps?

Reply:
`fu_backend_get_device_parent_with_kind` is probably even better.

Reply:
Yup that's a better name for what it does.

Reply:
Maybe have a fu_backend_get_device_parent additionally (without the kind hint) -- that'll improve readability too.

Reply:
Narrator: he did that.